src/main/java/thePackmaster/cards/startuppack/Improvise.java

```diff
@@ -0,0 +1,41 @@
+package thePackmaster.cards.startuppack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.DiscardSpecificCardAction;
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.common.GainBlockAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import thePackmaster.cards.AbstractPackmasterCard;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class Improvise extends AbstractStartUpCard {
+    public final static String ID = makeID("Improvise");
+
+    public Improvise() {
+        super(ID, 0, CardType.SKILL, CardRarity.UNCOMMON, CardTarget.SELF);
+        this.baseBlock = this.block = 5;
+        this.isInnate = true;
+    }
+
+    @Override
+    public void use(AbstractPlayer abstractPlayer, AbstractMonster abstractMonster) {
+        int cardsDiscarded = -1;
```

Comment:
Why start at -1?

Reply:
because i believe it still counts itself being in your hand, so it starts at -1 instead of 0 to account for itself.
I realize now though that this may then work weirdly when it is not played from hand but some other way...

Reply:
Fixed it by adding a c != this to the if statement